audio-worklet/bitcrusher-test.html

```diff
@@ -16,171 +16,184 @@
+        let context = new OfflineAudioContext(2, 44100, 44100);
+        const reduction = 1;
+        const bitDepth = 24;
+        const bufferSize = 512;
+        const channels = 1;
+        let oscillator = new OscillatorNode(context);
+        let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+        let bitcrusher = new Bitcrusher(
+            context, {bufferSize, channels, bitDepth, reduction});
+        // Accomodate for script processor latency by delaying the oscillator.
+        let delay = new DelayNode(context, {
+            delayTime: (bufferSize / context.sampleRate)
+        });
+
+        oscillator.connect(bitcrusher.input);
+        bitcrusher.output.connect(merger, 0, 0);
+        oscillator.connect(delay).connect(merger, 0, 1);
+        merger.connect(context.destination);
+        oscillator.start();
+
+        // When audio buffer is ready, verify bitcrushed samples are unaltered.
+        context.startRendering()
+            .then((buffer) => {
+              let bitcrusherOutput = buffer.getChannelData(0);
+              let originalOutput = buffer.getChannelData(1);
+
+              // Allow for fractional error beyond audible perception. This 
+              // error occurs because any sample passing through bitcrusher will 
+              // undergo manipulation in Math.round() and will therefore be 
+              // represented by a new floating point number that differs
+              // slightly from the original.In tested samples and at a sample 
+              // rate of 44,1k, the maximum observed error is 
+              // 6.183981895446777e-7.
+              const permittedSampleError = 6.1840e-7;
+              let maxError = 0;
+              for (let i = 0; i < bitcrusherOutput.length; i++) {
+                let diff = Math.abs(bitcrusherOutput[i] - originalOutput);
+                maxError = Math.max(diff, maxError);  
+              }
+              console.assert(
+                  maxError < permittedSampleError,
+                  'The maximum error between bitcrusher and original output, ' +
+                      maxError + ', exceeds the maximum permitted error, ' +
+                      permittedSampleError);
+
+              // Verify samples from unaltered oscillator match samples from
+              // bitcrushed oscillator with non-information reducing parameters.
+              for (let i = 0; i < originalOutput.length; i++) {
+                const crushedSample = bitcrusherOutput[i];
+                const originalSample = originalOutput[i];
+                const diff = Math.abs(originalSample - crushedSample);
+                if (i < bufferSize) {
+                  console.assert(
+                      bitcrusherOutput[i] === 0,
+                      'Bitcrusher sample at ' + i + 'expected to be 0 but it'
+                      + ' was ' + bitcrusherOutput[i]);
+                }
+                console.assert(
```

Comment:
Shouldn't this be in the "else" part of the "if" in line 113?  Otherwise this (redundantly) tests the first `bufferSize` samples twice.